esp/src/eclwatch/GraphWidget.js

```diff
@@ -71,6 +71,27 @@ define([
             this.calcColumns();
         },
 
+        query: function (query, options) {
+            var sortSet = options && options.sort;
+            if (sortSet) {
+                this.data.sort(typeof sortSet == "function" ? sortSet : function (a, b) {
+                    for (var sort, i = 0; sort = sortSet[i]; i++) {
+                        var aValue = a[sort.attribute];
+                        var bValue = b[sort.attribute];
+                        // valueOf enables proper comparison of dates
```

Comment:
Not in this case as its explaining why "valueOf" is being used - which is very different to leaving commented code inline or having comments which are clearly not needed.  
OUr rule of thumb is to "comment on the exceptional".